src/language/dafnyInstallation.ts

```diff
@@ -212,25 +213,29 @@ export class DafnyInstaller {
       and restart VSCode, which may reinstall Dafny.`);
       return false;
     }
-    try {
-      const process = await this.execLog('javac -version');
-      if(!(/javac \d+\.\d+/.exec(process.stdout))
-         && !(/javac \d+\.\d+/.exec(process.stderr))) {
-        throw '';
+    const configuredVersion = await getConfiguredVersion(this.context);
+    if(versionToNumeric(configuredVersion) < versionToNumeric('3.9.0')) {
+      try {
+
+        const process = await this.execLog('javac -version');
+        if(!(/javac \d+\.\d+/.exec(process.stdout))
+          && !(/javac \d+\.\d+/.exec(process.stderr))) {
+          throw '';
+        }
+      } catch(error: unknown) {
+        const errorMsg = error === '' ? 'Javac not found' : `${error}`;
+        this.writeStatus(`${errorMsg}. Please install a valid JDK`
```

Comment:
```suggestion
        this.writeStatus(`${errorMsg}. Javac is needed because you use a version of Dafny older than 3.9.0. Please install a valid JDK`
```